eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,115 @@
+  [Parameter(Mandatory=$false)][string]$BranchName
+)
+
+function TestPathThrow($Path) {
+  if(!(Test-Path $Path)) {
+   LogError "$Path not found"
+   exit
+  }
+}
+
+Write-Information "PS Script Root is: $PSScriptRoot"
+Write-Information "ArtifactName is: $ArtifactName"
+Write-Information "ReleaseVersion is: $ReleaseVersion"
+Write-Information "ServiceDirectoryName is: $ServiceDirectoryName"
+Write-Information "PatchVersion is: $PatchVersion"
+
+if($null -eq $BranchName) {
+  $BranchName = "release/$ArtifactName.ToLower()"
+}
+Write-Information "BranchName is: $branchName"
+
+$RepoRoot = Resolve-Path "${PSScriptRoot}..\..\.."
+$EngDir = Join-Path $RepoRoot "eng"
+$EngCommonScriptsDir = Join-Path $EngDir "common" "scripts"
+$SdkDirPath = Join-Path $RepoRoot "sdk"
+$ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
+$ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
+$ArtifactPomFile = Join-Path $ArtifactDirPath "pom.xml"
+$ReleaseTag = -join($ArtifactName, "_", $ReleaseVersion)
+$TestResourcesFilePath = Join-Path $ServiceDirPath "test-resources.json"
+$GroupId = "com.azure"
+$EngVersioningDir = Join-Path $EngDir "versioning"
+$SetVersionFilePath = Join-Path $EngVersioningDir "set_versions.py"
+$UpdateVersionFilePath = Join-Path $EngVersioningDir "update_versions.py"
+
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $EngDir
+TestPathThrow -Path $EngCommonScriptsDir
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $ArtifactPomFile
+TestPathThrow -Path $SdkDirPath
+TestPathThrow -Path $SetVersionFilePath
+TestPathThrow -Path $UpdateVersionFilePath
+
+## Creating a new branch
+git checkout -b $BranchName upstream/main
+
+try {
+  ## Hard reseting it to the contents of the release tag.
+  ## Fetching all the tags from the upstream branch
+  $CmdOutput = git fetch upstream --tags
+  $cmdOutput = git restore --source $ReleaseTag -W -S $ArtifactDirPath
+  if($LASTEXITCODE -ne 0) {
+    LogError "Could not restore the changes for release tag $ReleaseTag"
+    exit
+  }
+
+  if(Test-Path $TestResourcesFilePath) {
+    $cmdOutput = git restore --source $ReleaseTag -W -S $TestResourcesFilePath
+  }
+
+  ## Commit these changes.
+  git commit -a -m "Reset changes to the patch version."
+  
+  ## Create the patch release
+  python $SetVersionFilePath --bt client --new-version $PatchVersion --ar $ArtifactName --gi $GroupId
+  if($LASTEXITCODE -ne 0) {
+    LogError "Could not set the patch version."
+    exit
+  }
+  
+  python $UpdateVersionFilePath --ut library --bt client --tf $ArtifactPomFile --sr
+    if($LASTEXITCODE -ne 0) {
+    LogError "Could not update the versions in the pom files."
+    exit
+  }
+  
+  & "$EngCommonScriptsDir/Update-ChangeLog.ps1" -ReplaceLatestEntryTitle $False -Unreleased $False -Version $PatchVersion -ServiceDirectory $ServiceDirectoryName -PackageName $ArtifactName
+  
+  git add $RepoRoot
+  git commit -m "Updating the SDK dependencies for $ArtifactName"
```

Comment:
Printing the output of `git diff` is useful for users who run the script to ensure the changes were what they expected them to be.

Reply:
Do we really want the script to do it? Also, since we are committing the changes - would it not be more pragmatic for them to have to see it in a PR as called out? Git diff is mostly used for diffing non staged changes - no?